Corona-Warn-App/src/main/java/de/rki/coronawarnapp/service/submission/SubmissionService.kt

```diff
@@ -65,20 +65,8 @@ object SubmissionService {
     }
 
     fun containsValidGUID(scanResult: String): Boolean {
-        if (scanResult.length > SubmissionConstants.MAX_QR_CODE_LENGTH ||
-            scanResult.count { it == SubmissionConstants.GUID_SEPARATOR } != 1 ||
-            !matchGUIDPattern(scanResult)
-        )
-            return false
-
-        val potentialGUID = extractGUID(scanResult)
-
-        return !(potentialGUID.isEmpty() || potentialGUID.length > SubmissionConstants.MAX_GUID_LENGTH)
-    }
-
-    private fun matchGUIDPattern(scanResult: String): Boolean {
-        val pattern = SubmissionConstants.QR_CODE_REGEX
-        return pattern.toRegex().matches(scanResult)
+        val scanResult = ScanResult(scanResult)
```

Comment:
Close, but go even further, i.e. in `SubmissionViewModel.validateAndStoreTestGUID` start working with `ScanResult`.

The core idea is to stop passing `String` and start passing `ScanResult` as early as possible.

Reply:
Once refactored, move all related scan result tests into their own `ScanResultTest` class.